client/src/components/DatetimeField/DatetimeField.js

```diff
@@ -55,7 +55,7 @@ class DatetimeField extends DateField {
     let isoTime = '';
     if (localTime) {
       // Input value can be in local format 'L' or ISO format
-      const timeObject = this.moment(localTime, [localFormat, moment.ISO_8601]);
+      const timeObject = this.moment(localTime, [localFormat, 'L', moment.ISO_8601]);
```

Comment:
What does the 'L' do?

Reply:
Tests were failing here too cause they expect a date (not a date time) to be converted correctly. Might not be necessary since we're not changing moment, but I'd be surprised if tests weren't failing without these changes regardless.